### PR TITLE
Remove custom eviction policy class

### PR DESCRIPTION
**Pull request by kalaiyarasiganeshalingam**
## Purpose
$Subject

Fixes: [#1027](https://github.com/ballerina-platform/ballerina-standard-library/issues/1027)

### CODE REVIEW

File: cache-ballerina/cache.bal [hunk 1]
```diff
@@ -22,17 +22,22 @@ import ballerina/time;
...
 # + cleanupIntervalInSeconds - Interval of the timer task, which will clean up the cache
 public type CacheConfig record {|
     int capacity = 100;
     float evictionFactor = 0.25;
+    EvictionPolicy evictionPolicy = LRU;
     int defaultMaxAgeInSeconds = -1;
     int cleanupIntervalInSeconds?;
 |};
 
+public const string LRU = "leastRecentlyUse";
```
Comment (ldclakmal) [inline cache-ballerina/cache.bal:38]:
```suggestion
public const string LRU = "LeastRecentlyUsed";
```

Comment (kalaiyarasiganeshalingam) [reply cache-ballerina/cache.bal]: Eviction policy changed to Enum

File: cache-ballerina/cache.bal [hunk 2]
```diff
@@ -61,20 +67,21 @@ public class Cache {
...
-    private AbstractEvictionPolicy evictionPolicy;
+    private EvictionPolicy evictionPolicy;
     private float evictionFactor;
     private int defaultMaxAgeInSeconds;
+    private LinkedList linkedList;
 
     # Called when a new `cache:Cache` object is created.
     #
     # + cacheConfig - Configurations for the `cache:Cache` object
     # + evictionPolicy - The policy, which defines the cache eviction algorithm
```
Comment (ldclakmal) [inline cache-ballerina/cache.bal:78]:
```suggestion
```

File: cache-ballerina/cache.bal [hunk 3]
```diff
@@ -22,17 +22,23 @@ import ballerina/time;
...
 # + cleanupIntervalInSeconds - Interval of the timer task, which will clean up the cache
 public type CacheConfig record {|
     int capacity = 100;
     float evictionFactor = 0.25;
+    EvictionPolicy evictionPolicy = LRU;
     int defaultMaxAgeInSeconds = -1;
     int cleanupIntervalInSeconds?;
 |};
 
+public enum EvictionPolicy {
```
Comment (ldclakmal) [inline cache-ballerina/cache.bal:38]: Shall we add documentation for this enum?

Comment (kalaiyarasiganeshalingam) [reply cache-ballerina/cache.bal:39]: Added.

**Review by ldclakmal — APPROVED** [review]
LGTM

**Pull request merged**